airflow/models/connection.py

```diff
@@ -212,7 +212,9 @@ def get_uri(self) -> str:
                 self.conn_type,
             )
 
-        uri = f"{str(self.conn_type).lower().replace('_', '-')}://"
+        # SQLAlchemy expects postgresql:// as scheme
+        scheme = "postgresql" if self.conn_type == "postgres" else self.conn_type
+        uri = f"{str(scheme).lower().replace('_', '-')}://"
```

Comment:
```suggestion
        self.conn_type == "postgres":
            scheme = "postgresql"
        else:
            scheme = self.conn_type.lower().replace("_", "-")
        uri = f"{scheme}://"
```

No need to do the lower-replace combo on the “postgresql” literal.

Reply:
Sure - was aiming for minimum extra logic, but this is also fine :+1: